src/services/room-service.js

```diff
@@ -140,13 +146,28 @@ export default class RoomService {
         joinedOn: new Date()
       };
 
-      await this.roomStore.updateOne(
-        { _id: invitation.roomId },
-        { $push: { members: newMember } },
-        { session }
-      );
-
-      await this.roomInvitationStore.deleteOne({ _id: invitation._id }, { session });
+      let lock;
+
+      try {
+        try {
+          lock = await this.roomLockStore.takeLock(invitation.roomId);
+        } catch {
+          throw new BadRequest();
+        }
+
+        const room = await this.roomStore.findOne({ '_id': invitation.roomId, 'members.userId': newMember.userId });
+        if (!room) {
```

Comment:
`if (!room) {` can create misunderstanding here, because the room itself dues actually exist.

I would maybe call the variable `roomContainingNewMember` to make it a bit clearer.

Reply:
Renamed